shared_model/cryptography/hash_providers/sha3_256.hpp

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_SHARED_MODEL_SHA3_256_HPP
+#define IROHA_SHARED_MODEL_SHA3_256_HPP
+
+#include "common/types.hpp"
+#include "cryptography/ed25519_sha3_impl/internal/sha3_hash.hpp"
+
+namespace shared_model {
+  namespace crypto {
+    class Sha3_256 {
```

Comment:
Why do you need this class at all?

Reply:
In current API different hashing algorithms might be used. To enable use of sha3_256 this class is needed. We already had class for sha3_512

Reply:
We use it as a default template parameter for Hashable:
```typename HashProvider = shared_model::crypto::Sha3_256```